src/Catalyst.Core.Lib/Cryptography/IsaacRandom.cs

```diff
@@ -0,0 +1,239 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using Catalyst.Abstractions.Cryptography;
+using Nethereum.Hex.HexConvertors.Extensions;
+
+namespace Catalyst.Core.Lib.Cryptography
+{
+    /// <summary>
+    /// The Isaac Random
+    /// See www.rosettacode.org/wiki/The_ISAAC_Cipher
+    /// </summary>
+    public sealed class IsaacRandom : IDeterministicRandom
+    {
+        // external results 
+        private readonly uint[] _randRsl = new uint[256];
+        private uint _randCnt;
+
+        // internal state 
+        private readonly uint[] _mm = new uint[256];
+        private uint _aa;
+        private uint _bb;
+        private uint _cc;
+
+        private void Isaac()
+        {
+            uint i;
+            _cc++; // _cc just gets incremented once per 256 results 
+            _bb += _cc; // then combined with _bb 
+
+            for (i = 0; i <= 255; i++)
+            {
+                var x = _mm[i];
+                switch (i & 3)
+                {
+                    case 0:
+                        _aa ^= _aa << 13;
+                        break;
+                    case 1:
+                        _aa ^= _aa >> 6;
+                        break;
+                    case 2:
+                        _aa ^= _aa << 2;
+                        break;
+                    case 3:
+                        _aa ^= _aa >> 16;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+
+                _aa = _mm[(i + 128) & 255] + _aa;
+                var y = _mm[(x >> 2) & 255] + _aa + _bb;
+                _mm[i] = y;
+                _bb = _mm[(y >> 10) & 255] + x;
+                _randRsl[i] = _bb;
+            }
+        }
+
+        void Mix(ref uint a, ref uint b, ref uint c, ref uint d, ref uint e, ref uint f, ref uint g, ref uint h)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method has 8 parameters, which is greater than the 7 authorized.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)